Improve the kernel debug setup again.

AShiningRay opened the pull request:
We now have all those debug flags from before, but now transformed
into kernel configs that can be toggled in mk2000_defconfig,
making things much easier when only some parts of the kernel need
to be debugged.

AShiningRay committed:
Improve the kernel debug setup again.

We now have all those debug flags from before, but now transformed
into kernel configs that can be toggled in mk2000_defconfig,
making things much easier when only some parts of the kernel need
to be debugged.

stendro commented:
I see a few issues. You're not using 'subdir-ccflags' so it will not be the same, and in the case of USB it won't do anything since all the code is in subdirs.
There were also a few specific omissions I made to avoid some extremely noisy logging, although that probably matters less when we only enable one or two of these.

AShiningRay committed:
Small improvement to the debug layout.

As @stendro pointed out, some -DDEBUG flags need to be applied
to all subdirs where the Makefile is located, that's the case for
usb for example. Only the important ones were turned into subdir
flags while others were in-depth enough to not need them, as well
as the subdirs that Lg's trio didn't even use. As for the file
omissions to reduce log spam, i prefer to let them in seeing as
power_supply debug also tendds to spam a lot of _changed() calls,
and we also won't be as impacted by them unless we enable all at
the same time so we can keep the amount of changes to a minimum.

AShiningRay commented:
Alright, should be more in line with the previous layout now. If there's anything else that needs to be done, please tell.

stendro commented:
A bit funny that we have the separate USB host option when everything is enabled by the regular USB anyway, the host part seems like a less likely debug candidate too, for us.

AShiningRay commented:
Yeah, i only left it there in the hopes it proves useful to debug OTG-related problems later and without the need to enable debugging on the whole USB stack and risking lots of useless entries.
